[webkitperl] Define package webkitdirs and export functions used in other modules

dpino opened the pull request:
#### ecde243930f2bd6202905b67322b95a38e6b9119
<pre>
[webkitperl] Define package webkitdirs and export functions used in other modules
<a href="https://bugs.webkit.org/show_bug.cgi?id=243454">https://bugs.webkit.org/show_bug.cgi?id=243454</a>

Reviewed by Ross Kirsling.

Module &apos;webkitdirs.pm&apos; needs a package name definition, otherwise the export function mechanism
doesn&apos;t work.

What was happening until now was that all functions defined in &apos;webkitdirs.pm&apos; were exported into
the &apos;main&apos; namespace. A module living in the &apos;main&apos; namespace (that is, a module without a package
name), could use all functions defined in &apos;webkitdirs.pm&apos;. But a module with a package name would
need to append the namespace &apos;main&apos; to call a function defined in &apos;webkitdirs.pm&apos;.

Once the package name for &apos;webkitdirs&apos; was defined it was also necessary to export a few more
functions that were used in other modules.

* Tools/Scripts/webkitdirs.pm: Define module name and export functions used from other modules.
* Tools/Scripts/webkitperl/FeatureList.pm: Define module name.
* Tools/Scripts/webkitperl/prepare-ChangeLog_unittest/parser_unittests.pl:
(convertAbsolutePathToRelativeUnixPath): Function &apos;unixPath&apos; is not defined in LoadAsModule package
but VCSUtils.
* Tools/Scripts/do-webcore-rename: Use &apos;File::Basename&apos;.
* Tools/Scripts/package-root:
* Tools/Scripts/run-javascriptcore-tests:
* Tools/Scripts/run-regexp-tests:
* Tools/Scripts/run-sunspider:
* Tools/Scripts/set-webkit-configuration:
* Tools/Scripts/sunspider-compare-results:
* Tools/Scripts/update-webkit-libs-jhbuild:
* Tools/Scripts/webkit-build-directory:
* Tools/Scripts/webkitperl/BuildSubproject.pm:

Canonical link: <a href="https://commits.webkit.org/253110@main">https://commits.webkit.org/253110@main</a>
</pre>

rkirsling reviewed (approved):
r=me once things are green

dpino committed:
[webkitperl] Define package webkitdirs and export functions used in other modules
https://bugs.webkit.org/show_bug.cgi?id=243454

Reviewed by Ross Kirsling.

Module 'webkitdirs.pm' needs a package name definition, otherwise the export function mechanism
doesn't work.

What was happening until now was that all functions defined in 'webkitdirs.pm' were exported into
the 'main' namespace. A module living in the 'main' namespace (that is, a module without a package
name), could use all functions defined in 'webkitdirs.pm'. But a module with a package name would
need to append the namespace 'main' to call a function defined in 'webkitdirs.pm'.

Once the package name for 'webkitdirs' was defined it was also necessary to export a few more
functions that were used in other modules.

* Tools/Scripts/webkitdirs.pm: Define module name and export functions used from other modules.
* Tools/Scripts/webkitperl/FeatureList.pm: Define module name.
* Tools/Scripts/webkitperl/prepare-ChangeLog_unittest/parser_unittests.pl:
(convertAbsolutePathToRelativeUnixPath): Function 'unixPath' is not defined in LoadAsModule package
but VCSUtils.
* Tools/Scripts/do-webcore-rename: Use 'File::Basename'.
* Tools/Scripts/package-root:
* Tools/Scripts/run-javascriptcore-tests:
* Tools/Scripts/run-regexp-tests:
* Tools/Scripts/run-sunspider:
* Tools/Scripts/set-webkit-configuration:
* Tools/Scripts/sunspider-compare-results:
* Tools/Scripts/update-webkit-libs-jhbuild:
* Tools/Scripts/webkit-build-directory:
* Tools/Scripts/webkitperl/BuildSubproject.pm:

Canonical link: https://commits.webkit.org/253110@main
